src/pages/Config.js

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+export default function Config() {
+  const [gateWidth, setGateWidth] = useState(2500);
+  const [gateHeight, setGateHeight] = useState(2000);
+  const [messages, setMessages] = useState([]);
+  const [qm, setQm] = useState(5);
+  const [show, setShow] = useState(Math.random());
+
+  useEffect(() => {
+    setQm(
+      ((gateWidth * gateHeight) / 1000000).toLocaleString(undefined, {
+        maximumFractionDigits: 2,
+        minimumFractionDigits: 2,
+      })
+    );
+  }, [show]);
```

Comment:
```suggestion
useEffect(() => {
    if (isNaN(gateHeight) || isNaN(gateWidth)) setQm("--");
    else
      setQm(
        ((+gateWidth * +gateHeight) / 1000000).toLocaleString(undefined, {
          maximumFractionDigits: 2,
          minimumFractionDigits: 2,
        })
      );
  }, [gateHeight, gateWidth]);
```
Add a check for valid height and width values and display some kind of invalid input state "--".
Additionally, the useEffect hook responds to changes of said height and width values instead of an intermediate show state which merely functioned as a proxy for gateWidth and gateHeight.